appknox/reports.go

```diff
@@ -4,139 +4,67 @@ import (
 	"context"
 	"errors"
 	"fmt"
-	"io"
-	"net/http"
-	"os"
-	"path/filepath"
-	"strings"
+	"strconv"
+	"time"
 )
 
 type ReportsService service
 
-type Report struct {
-	URL string `json:"url"`
+type HIPAAPreferences struct {
+	ShowHIPPA   bool `json:"value"`
+	IsInherited bool `json:"is_inherited"`
 }
-
-type ReportResult struct {
-	ID          int    `json:"id"`
-	GeneratedOn string `json:"generated_on"`
-	Language    string `json:"language"`
-	Progress    int    `json:"progress"`
-	Rating      string `json:"rating"`
+type PCIDSSPreferences struct {
+	ShowPCIDSS  bool `json:"value"`
+	IsInherited bool `json:"is_inherited"`
 }
 
-// GenerateReport generates a report for the specified file.
-func (s *ReportsService) GenerateReport(ctx context.Context, fileID int) (*ReportResult, error) {
-	url := fmt.Sprintf("api/v2/files/%d/reports", fileID)
-	req, err := s.client.NewRequest("POST", url, nil)
-	if err != nil {
-		return nil, err
-	}
-
-	var resp ReportResult
-
-	res, err := s.client.Do(ctx, req, &resp)
-	if err != nil {
-		if res.StatusCode == 400 {
-			return nil, errors.New("A report is already being generated or scan is in progress. Please wait.")
-		}
-		if res.StatusCode == 404 {
-			return nil, errors.New("File not found")
-		}
-		return nil, err
-	}
-
-	return &resp, nil
+type ReportPreferences struct {
+	ShowAPIScan         bool              `json:"show_api_scan"`
+	ShowManualScan      bool              `json:"show_manual_scan"`
+	ShowStaticScan      bool              `json:"show_static_scan"`
+	ShowDynamicScan     bool              `json:"show_dynamic_scan"`
+	ShowIgnoredAnalyses bool              `json:"show_ignored_analyses_scan"`
+	PCIDSSPreferences   PCIDSSPreferences `json:"show_hipaa"`
+	HIPAAPreferences    HIPAAPreferences  `json:"show_pcidss"`
 }
 
-// FetchReportResult it will fetch report result by result id.
-func (s *ReportsService) FetchReportResult(ctx context.Context, reportID int) (*ReportResult, error) {
-	url := fmt.Sprintf("api/v2/reports/%d", reportID)
-	req, err := s.client.NewRequest("GET", url, nil)
-	if err != nil {
-		return nil, err
-	}
-
-	var resp ReportResult
-
-	_, err = s.client.Do(ctx, req, &resp)
-	if err != nil {
-		return nil, err
-	}
+type ReportResult struct {
+	ID                int               `json:"id"`
+	GeneratedOn       *time.Time        `json:"generated_on"`
+	Language          string            `json:"language"`
+	Progress          int               `json:"progress"`
+	Rating            string            `json:"rating"`
+	ReportPreferences ReportPreferences `json:"preferences"`
```

Comment:
I think we should remove report preferences altogether. We don't need it. It'll just make report rendering difficult.